calculator.c

```diff
@@ -0,0 +1,117 @@
+#include <stdio.h>
+#include <ctype.h>
+#include <stdlib.h>
+#define MAX_SIZE 100
+
+int operation(int a, int b, char op)
+{
+    switch (op)
+    {
+    case '+':
+        return a + b;
+    case '-':
+        return a - b;
+    case '*':
+        return a * b;
+    case '/':
+        if (b == 0)
+        {
+            printf("Error: Division by zero.\n");
+            exit(1);
+        }
+        return a / b;
+    default:
+        return 0;
+    }
+}
+
+int precedence(char op)
+{
+    if (op == '+' || op == '-')
+        return 1;
+    if (op == '*' || op == '/')
+        return 2;
+    return 0;
+}
+
+int evaluate(char *exp)
+{
+    int stack1[MAX_SIZE], top1 = -1;
+    char stack2[MAX_SIZE], top2 = -1;
+    int i = 0;
+    int lastOperator = 1;
+    while (exp[i] != '\0' && exp[i] != '\n')
+    {
+        if (exp[i] == ' ')
+        {
+            i++;
+            continue;
+        }
+        if (exp[i] == '-' && (i == 0 || lastOperator))
+        {
+            i++;
+            int num = 0;
+            while (isdigit(exp[i]))
```

Comment:
Avoid using inbuilt functions unless it is system calls, please change the built-in functions to user implementation function. take care at all places.